tools/tool.sh

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+set -o errexit -o nounset -o pipefail
+
+tool="${PWD}/$1"
```

Comment:
does it differ from our template on purpose? why pass the tool as an argument rather than name the script after the tool?

Reply:
Copied from silo. Is there a better approach that the CLI template can set up?

Reply:
https://github.com/aspect-build/aspect-workflows-template/tree/main/%7B%7B%20.ProjectSnake%20%7D%7D is the template stamped out by `aspect init` and contains my recommendations. I think we should probably change silo too, though last time I looked it was not trivial

Reply:
I think the approach I have in this repo is more ideal for what I need. For example, by declaring the `terraform` macro in the deployments directly, I can add the dependency on building the thing that's being deployed:

```starlark
terraform(
    name = "terraform",
    data = [
        "//src:cloudfunction",
    ],
)
```

```
(cd deployment/environments/dev && bazel run -c opt terraform -- apply)
```

Otherwise I need to be careful to run a `bazel build` first.